### PR TITLE
Add configure, configure Makevars.in -> Makevars

### PR DESCRIPTION
**Pull request by mlondschien**

### CODE REVIEW

File: changeforest-r/src/Makevars.in [hunk 1]
```diff
@@ -1,3 +1,4 @@
+TARGET ?= @RUST_TARGET@
 TARGET_DIR = ./rust/target
 LIBDIR = $(TARGET_DIR)/release
```
Comment (Ilia-Kosenkov) [inline changeforest-r/src/Makevars.in:3]: I think you are missing this now
```suggestion
LIBDIR = $(TARGET_DIR)/$(TARGET)/release
```

**Comment by Ilia-Kosenkov** [comment]
I think some one else is missing, unable to cross-compile. Lemme check

**Comment by Ilia-Kosenkov** [comment]
@eitsupi, hi, we are dealing with a cross-compilation issue. Could you please help us figure out cross-compilation from Max OS `x86_64` to `aarch64`? I used `poalrs-r` as an example, but I still do not understand how to build a package without linking to R library, which in this case causes an architecture mismatch (R session links to `x86_64` library while cargo artifact is `aarch64`)

**Comment by eitsupi** [comment]
I don't know how to cross compile R packages.
Regarding conda-forge, packages like dplyr have binaries for arm64, so why not use them as a reference?

The only thing I know a little about is cross-compilation in Rust. I'm sure you can find a lot of information about it on the internet.

**Comment by yutannihilation** [comment]
Okay, so, if even polars is not cross-compile-aware, I guess there's no Rust-powered R package that does it correctly. In my understanding, extendr never aimed at cross-compiling for arm64 macOS, so it's no surprise that this doesn't work. Anyone would be welcomed to explore this frontier. But, IMHO, cross-compiling is a temporary solution until the arm64 macOS runner is generally available, so I'm not sure if it's really worth the effort.

That being said, here's a good news (or bad news) that we might get a reason that forces us to think about cross-compiling seriously.

https://github.com/r-universe-org/help/issues/97#issuecomment-1880221054

**Comment by eitsupi** [comment]
What is possible with `prqlr` (and `polars`) is to cross-compile the Rust library in advance and use that binary when installing the R package to avoid the cargo build when installing the R package.
https://github.com/eitsupi/prqlr/blob/76c68dda761c0b5f5783b03154c20b6127b306b1/configure#L50-L66

So if a cross-compilation method has been established for C, it would be possible to build the R package by cross-compiling Rust and C in this way, respectively.

As I mentioned above, some packages such as dplyr are available on arm64 Linux and macOS on conda-forge, so it is not impossible to build R packages using Rust, I think?

**Comment by yutannihilation** [comment]
Yeah, I believe it's definitely possible and I bet it's just specifying `target` correctly.

**Comment by eitsupi** [comment]
As for Linux, I believe we can build the R package by cross-compiling the Rust library on the host machine and then running arm64 R on Docker with qemu.
But not sure about macOS.

Since cargo-cross seems to support cross-compilation of packages other than Linux, it may be possible to cross-compile R packages in the same way, but I have not tried it.
(Incidentally, since there is no R on the Docker image used by cargo-cross, it cannot be used for Linux as it is.)
https://github.com/cross-rs/cross

**Comment by yutannihilation** [comment]
https://github.com/r-universe/workflows/blob/9355082e241f26496479c1808caed5e15a3d9861/.github/workflows/build.yml#L262-L267
https://github.com/r-universe-org/build-and-check/blob/v1/macos-cross/action.yml

**Comment by yutannihilation** [comment]
Confirmed that specifying `--target` makes it possible to cross-compile for arm64 macOS at least on R-universe. I'm not interested in conda so I'd stop here, but hope this helps to some extent.

diff: https://github.com/yutannihilation/string2path/compare/912ae92..9576797
log: https://github.com/r-universe/yutannihilation/actions/runs/7443083907/job/20247512893#step:7:56

**Comment by mlondschien** [comment]
I get the same error locally:
```
    Finished release [optimized] target(s) in 1m 57s
if [ "" != "true" ]; then \
                rm -Rf /Users/mlondschien/code/changeforest/changeforest-r/src/.cargo && \
                rm -Rf ./rust/target/release/build; \
        fi
arm64-apple-darwin20.0.0-clang -dynamiclib -Wl,-headerpad_max_install_names -undefined dynamic_lookup -single_module -multiply_defined suppress -L/Users/mlondschien/mambaforge/envs/r-test/lib/R/lib -Wl,-dead_strip_dylibs -Wl,-pie -Wl,-headerpad_max_install_names -Wl,-dead_strip_dylibs -Wl,-rpath,/Users/mlondschien/mambaforge/envs/r-test/lib -L/Users/mlondschien/mambaforge/envs/r-test/lib -o changeforest.dylib entrypoint.o -L./rust/target/release -lchangeforestr -L/Users/mlondschien/mambaforge/envs/r-test/lib/R/lib -lR -Wl,-framework -Wl,CoreFoundation
ld: library not found for -lchangeforestr
arm64-apple-darwin20.0: error: linker command failed with exit code 1 (use -v to see invocation)
make: *** [changeforest.dylib] Error 1
ERROR: compilation failed for package ‘changeforest’
* removing ‘/Users/mlondschien/mambaforge/envs/r-test/lib/R/library/changeforest’
* restoring previous ‘/Users/mlondschien/mambaforge/envs/r-test/lib/R/library/changeforest’
```

If I replace `-lchangeforestr` with `-lchangeforest` in the `Makevars`, I get


```
arm64-apple-darwin20.0.0-clang -dynamiclib -Wl,-headerpad_max_install_names -undefined dynamic_lookup -single_module -multiply_defined suppress -L/Users/mlondschien/mambaforge/envs/r-test/lib/R/lib -Wl,-dead_strip_dylibs -Wl,-pie -Wl,-headerpad_max_install_names -Wl,-dead_strip_dylibs -Wl,-rpath,/Users/mlondschien/mambaforge/envs/r-test/lib -L/Users/mlondschien/mambaforge/envs/r-test/lib -o changeforest.dylib entrypoint.o -L./rust/target/release -lchangeforest -L/Users/mlondschien/mambaforge/envs/r-test/lib/R/lib -lR -Wl,-framework -Wl,CoreFoundation
ld: warning: -pie being ignored. It is only used when linking a main executable
installing to /Users/mlondschien/mambaforge/envs/r-test/lib/R/library/00LOCK-changeforest-r/00new/changeforest/libs
** R
** byte-compile and prepare package for lazy loading
** help
*** installing help indices
** building package indices
** testing if installed package can be loaded from temporary location
Error: package or namespace load failed for ‘changeforest’ in dyn.load(file, DLLpath = DLLpath, ...):
 unable to load shared object '/Users/mlondschien/mambaforge/envs/r-test/lib/R/library/00LOCK-changeforest-r/00new/changeforest/libs/changeforest.dylib':
  dlopen(/Users/mlondschien/mambaforge/envs/r-test/lib/R/library/00LOCK-changeforest-r/00new/changeforest/libs/changeforest.dylib, 0x0006): symbol not found in flat namespace '_R_init_changeforest_extendr'
Error: loading failed
Execution halted
ERROR: loading failed
* removing ‘/Users/mlondschien/mambaforge/envs/r-test/lib/R/library/changeforest’
* restoring previous ‘/Users/mlondschien/mambaforge/envs/r-test/lib/R/library/changeforest’
```

**Comment by mlondschien** [comment]
I get the same error locally:
```
    Finished release [optimized] target(s) in 1m 57s
if [ "" != "true" ]; then \
                rm -Rf /Users/mlondschien/code/changeforest/changeforest-r/src/.cargo && \
                rm -Rf ./rust/target/release/build; \
        fi
arm64-apple-darwin20.0.0-clang -dynamiclib -Wl,-headerpad_max_install_names -undefined dynamic_lookup -single_module -multiply_defined suppress -L/Users/mlondschien/mambaforge/envs/r-test/lib/R/lib -Wl,-dead_strip_dylibs -Wl,-pie -Wl,-headerpad_max_install_names -Wl,-dead_strip_dylibs -Wl,-rpath,/Users/mlondschien/mambaforge/envs/r-test/lib -L/Users/mlondschien/mambaforge/envs/r-test/lib -o changeforest.dylib entrypoint.o -L./rust/target/release -lchangeforestr -L/Users/mlondschien/mambaforge/envs/r-test/lib/R/lib -lR -Wl,-framework -Wl,CoreFoundation
ld: library not found for -lchangeforestr
arm64-apple-darwin20.0: error: linker command failed with exit code 1 (use -v to see invocation)
make: *** [changeforest.dylib] Error 1
ERROR: compilation failed for package ‘changeforest’
* removing ‘/Users/mlondschien/mambaforge/envs/r-test/lib/R/library/changeforest’
* restoring previous ‘/Users/mlondschien/mambaforge/envs/r-test/lib/R/library/changeforest’
```

If I replace `-lchangeforestr` with `-lchangeforest` in the `Makevars`, I get


```
arm64-apple-darwin20.0.0-clang -dynamiclib -Wl,-headerpad_max_install_names -undefined dynamic_lookup -single_module -multiply_defined suppress -L/Users/mlondschien/mambaforge/envs/r-test/lib/R/lib -Wl,-dead_strip_dylibs -Wl,-pie -Wl,-headerpad_max_install_names -Wl,-dead_strip_dylibs -Wl,-rpath,/Users/mlondschien/mambaforge/envs/r-test/lib -L/Users/mlondschien/mambaforge/envs/r-test/lib -o changeforest.dylib entrypoint.o -L./rust/target/release -lchangeforest -L/Users/mlondschien/mambaforge/envs/r-test/lib/R/lib -lR -Wl,-framework -Wl,CoreFoundation
ld: warning: -pie being ignored. It is only used when linking a main executable
installing to /Users/mlondschien/mambaforge/envs/r-test/lib/R/library/00LOCK-changeforest-r/00new/changeforest/libs
** R
** byte-compile and prepare package for lazy loading
** help
*** installing help indices
** building package indices
** testing if installed package can be loaded from temporary location
Error: package or namespace load failed for ‘changeforest’ in dyn.load(file, DLLpath = DLLpath, ...):
 unable to load shared object '/Users/mlondschien/mambaforge/envs/r-test/lib/R/library/00LOCK-changeforest-r/00new/changeforest/libs/changeforest.dylib':
  dlopen(/Users/mlondschien/mambaforge/envs/r-test/lib/R/library/00LOCK-changeforest-r/00new/changeforest/libs/changeforest.dylib, 0x0006): symbol not found in flat namespace '_R_init_changeforest_extendr'
Error: loading failed
Execution halted
ERROR: loading failed
* removing ‘/Users/mlondschien/mambaforge/envs/r-test/lib/R/library/changeforest’
* restoring previous ‘/Users/mlondschien/mambaforge/envs/r-test/lib/R/library/changeforest’
```

**Comment by yutannihilation** [comment]
If you specify `--target=aarch64-apple-darwin`, `LIBDIR` should be `./rust/target/aarch64-apple-darwin/release`, not `./rust/target/release`.

```
arm64-apple-darwin20.0.0-clang ..snip.. -L./rust/target/release -lchangeforestr ...
````

**Pull request merged**